pkg/monitor/lease.go

```diff
@@ -0,0 +1,137 @@
+package monitor
+
+import (
+	"fmt"
+	"net"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"github.com/milosgajdos/tenus"
+	"github.com/openshift/baremetal-runtimecfg/pkg/config"
+	"github.com/openshift/baremetal-runtimecfg/pkg/utils"
+	"github.com/sirupsen/logrus"
+)
+
+/*
+	After reading the cmd flags and before executing the keepalived-monitor:
+	1. Get the directory of args[2] = path_to_config (e.g. /etc/keepalived/keepalived.conf) [V]
+	2. Check if the there's a file named "monitor.conf" in that directory. If true, then: [V]
+		2.1. for [API interface, Ingress interface]: . [V]
+			2.1.1 Generate a unique MAC address
+				2.1.1.1. The first 24 bits of the MAC should be the same as the real interface mac address.
```

Comment:
I don't think we can use the real hardware address as part of the MAC generation. Otherwise the MAC is going to change depending on which system is running the monitor.